pkg/apis/core/validation/validation.go

```diff
@@ -4356,10 +4357,24 @@ func ValidateConfigMap(cfg *core.ConfigMap) field.ErrorList {
 		for _, msg := range validation.IsConfigMapKey(key) {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, msg))
 		}
+		// check if the value stored consists entirely of valid UTF-8-encoded runes
+		if !utf8.ValidString(cfg.Data[key]) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, "invalid byte sequence in UTF-8"))
+		}
+		// check if we have a duplicate key in the other bag
+		if _, isValue := cfg.BinaryData[key]; isValue {
+			allErrs = append(allErrs, field.Duplicate(field.NewPath("data").Key(key), "duplicate of key in binaryData"))
+		}
+		totalSize += len(value)
+	}
+	for key, value := range cfg.BinaryData {
+		for _, msg := range validation.IsConfigMapKey(key) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("binaryData").Key(key), key, msg))
+		}
 		totalSize += len(value)
 	}
 	if totalSize > core.MaxSecretSize {
-		allErrs = append(allErrs, field.TooLong(field.NewPath("data"), "", core.MaxSecretSize))
+		allErrs = append(allErrs, field.TooLong(field.NewPath(""), cfg, core.MaxSecretSize))
```

Comment:
don't pass "" as a field path.  That's just going to produce something useless.  At least pass `"(data + binaryData)"`

Reply:
heh, I asked him to revert an equivalent fake path "data|binaryData"... is the expectation not that field path should be able to be followed programatically? do we have a way to indicate "the whole object"?

Reply:
In JSONPath it's "$" or "." or "".  I would say the closest to our intent is "" for "the whole object" because fieldPath is optional in api.Status.

Reply:
Ugh, I hate this.  BUt I don't have anything better.  Leave a comment?